model/cover_graph.py

```diff
@@ -24,6 +24,41 @@ def renamed_edges(self, old_vertex, new_vertex):
             new_edges.append(edge)
         return new_edges
 
+    def get_labels(self, operations=['assignment', 'skip', 'if', 'while']):
+        return [vertex.label for vertex in self.vertices if vertex.operation in operations]
+
+    def get_variables(self):
+        to_return = set()
+        for edge in self.edges:
+            to_return |= edge.get_variables()
+        return to_return
+
+    def get_def_variables(self, vertex):
+        to_return = set()
+        for edge in self.edges:
+            if type(vertex) == int:
+                if edge.root_vertex.operation == 'assignment' and edge.root_vertex.label == vertex:
+                    to_return |= edge.operation.left_expression.get_variables()
+            else:
+                if edge.root_vertex.operation == 'assignment' and edge.root_vertex == vertex:
+                    to_return |= edge.operation.left_expression.get_variables()
+        return to_return
```

Comment:
A few things looks strange to me : `vertex` will be of type `Vertex` no ? In what kind of situation it could be an int ?
Also in my part I assumed that `edge.root_vertex.operation` contains `AssignmentNode` objects and not only the label. What is the correct answer ?

Reply:
My answer:
1. I consider the case when we give a label instead of a vertex. I was not sure about how you will build criterias. We can suppress the label part if it is useless.
2. The operation variable for Edge class is either a SkipNode or an AssignementNode instance. The operation variable for Vertex class is only a string in Vertex.possible_operations. Here I think we don't need to embarq all Node logic here because the eval function will be on edges. Maybe this is not the good way of building our logic.

Reply:
Ok I'll really fucked up in other PR's. I didn't see that you tought that vertex has operation node and not a string. Fixed proposed in #6.

Reply:
Ok for me